### ISSUE TITLE
🐛 BUG: Resolved asset paths incorrect for `pageUrlFormat: 'file'`

### ISSUE THREAD

**Issue by coelmay** [withastro/astro#2343]
### What version of `astro` are you using?

0.22.9

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Following @matthewp's https://github.com/withastro/astro/pull/2330 to fix #1338, the issue raised in https://github.com/withastro/astro/issues/1944#issuecomment-997132328 is now in all files, that `Astro.resolve` paths remain prefixed with a `../` e.g. in `/404.html`
```html
<link rel="stylesheet" href="../assets/global.02ce08a9.css">
```
instead of
```html
<link rel="stylesheet" href="assets/global.02ce08a9.css">
```
or
```html
<link rel="stylesheet" href="/assets/global.02ce08a9.css">
```

I have updated the demonstration repository to 0.22.9 and included the `dist` built locally which demonstrates the issue.

### Link to Minimal Reproducible Example

https://github.com/coelmay/astro-resolve-stylesheet-404-issue

**Comment by matthewp** [comment]
Yeah good point. I we are resolving those to the dev path but we should actually calculate the output path (because of pageUrlFormat: 'file') and use that in resolution instead.

**Comment by FredKSchott** [comment]
Fixed in v0.24.0 with the deprecation of `Astro.resolve()`

**Comment by coelmay** [comment]
Doesn't work with `--legacy-build` flag enabled though @FredKSchott. This intentional?

Two issues though:
1)
I find the naming of the CSS file very strange. Am using TailwindCSS so have
```html
<style global>
  @import "../styles/global.css";
</style>
```
and the file final file is `/assets/SiteScripts.eca8354e.css` (I have a component `SiteScripts.astro` which contains no CSS) but an `index.db66d5eb.css` file is also created that is empty and only referenced in the root `index.html`.
2)
The output generates a end tag `</link>`
```html
<link rel="stylesheet" href="/assets/SiteScripts.eca8354e.css"></link>
```
though [there is not end tag](https://html.spec.whatwg.org/multipage/semantics.html#the-link-element).
